manifests/repos.pp

```diff
@@ -8,7 +8,6 @@
   case $facts['os']['family'] {
     'RedHat', 'Linux': {
       $yumcode = $facts['os']['name'] ? {
-        'Amazon' => 'el7',
         'Fedora' => "f${facts['os']['release']['major']}",
         default  => "el${facts['os']['release']['major']}",
       }
```

Comment:
We haven't had Fedora releases for a long time. Simplify it to `$yumcode = "el${facts['os']['release']['major']}"` perhaps? Or even inline it in the `foreman::repos::yum` declaration a few lines below.